src/psyclone/psyir/nodes/reference.py

```diff
@@ -250,6 +250,24 @@ def next_access(self):
             return all_accesses[index+1].node
         return None
 
+    def update_symbols_from(self, table):
```

Comment:
Loop also has a symbol for its looping variable. We have a special case in "scoping_node.py::_refine_copy" for it. Could this responsibility be moved to a Loop "update_symbols_from" method?

Reply:
Great idea. Some code removed.